src/ConditionAnalyzer/Analyzer.php

```diff
@@ -203,7 +211,10 @@ private function hasValidQueryOperator($hash, $range = null)
         $validQueryOp = ComparisonOperator::isValidQueryDynamoDbOperator($hashConditionType);
 
         if ($validQueryOp && $range) {
-            $rangeConditionType = $this->getCondition($range)['type'] ?? null;
+            $rangeConditionType = null;
+            if($this->getCondition($range)) {
+                $rangeConditionType = $this->getCondition($range)['type'] ?? null;
+            }
```

Comment:
@thebatclaudio Actually, you should use the inline check as you had before. I tested locally and worked well, this new version does not work, because takes `null` as invalid operator for sort key.

Just check against `not null`: `$this->getCondition($range) !== null`, is the only condition to further check for operator type of range.

Reply:
Because the intention is to allow `Query` when there is no condition for range at all.

Reply:
Hi @nelson6e65,
sorry, my fault. I just fixed also this.

Reply:
No worries, great job, @thebatclaudio! :D